cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -52,14 +52,47 @@ const (
 	kubeScheduler         = "kube-scheduler"
 )
 
-// WriteStaticPodManifests builds manifest objects based on user provided configuration and then dumps it to disk
-// where kubelet will pick and schedule them.
-func WriteStaticPodManifests(cfg *kubeadmapi.MasterConfiguration) error {
+// CreateInitStaticPodManifestFiles will create and write to disk all static pod manifest files necessary in the kubeadm init phase
+// to establish the control plane, including eventually static pod manifest if necessary.
+func CreateInitStaticPodManifestFiles(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	initManifests := []string{kubeAPIServer, kubeControllerManager, kubeScheduler}
+
+	// Add etcd static pod spec only if external etcd is not configured
+	if len(cfg.Etcd.Endpoints) == 0 {
+		initManifests = append(initManifests, etcd)
+	}
+
+	return createStaticPodFiles(outDir, cfg, initManifests...)
+}
+
+// CreateAPIServerStaticPodManifestFile will create and write to disk the APIserver static pod manifest file.
+func CreateAPIServerStaticPodManifestFile(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	return createStaticPodFiles(outDir, cfg, kubeAPIServer)
+}
+
+// CreateControllerManagerStaticPodManifestFile will create and write to disk the controller manager static pod manifest file.
+func CreateControllerManagerStaticPodManifestFile(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	return createStaticPodFiles(outDir, cfg, kubeControllerManager)
+}
+
+// CreateSchedulerStaticPodManifestFile will create and write to disk the scheduler static pod manifest file.
+func CreateSchedulerStaticPodManifestFile(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	return createStaticPodFiles(outDir, cfg, kubeScheduler)
+}
+
+// CreateEtcdStaticPodManifestFile will create and write to disk the etcd static pod manifest file.
+func CreateEtcdStaticPodManifestFile(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	return createStaticPodFiles(outDir, cfg, etcd)
+}
+
+// getStaticPodSpecs returns all staticPodSpecs actualized to the context of the current MasterConfiguration
+// NB. this methods holds the information about how kubeadm creates static pod mainfests.
+func getStaticPodSpecs(cfg *kubeadmapi.MasterConfiguration) (map[string]v1.Pod, error) {
 
 	// TODO: Move the "pkg/util/version".Version object into the internal API instead of always parsing the string
 	k8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
 	if err != nil {
-		return err
+		return nil, err
```

Comment:
In some PR I have open to merge now, I've moved parsing of k8sVersion to outside of this func so that this doesn't have to return an error.